api/data_explorer/__main__.py

```diff
@@ -238,26 +217,38 @@ def _get_interval(max_field_value):
         return 1000000000000
 
 
-def _get_es_facets():
-    """Returns a dict from UI facet name to Elasticsearch facet object."""
+def _process_facets():
+    """Process facets to store a dict from UI facet name to UI facet description
+    and a dict of UI facet name to Elasticsearch facet object.
+    If there is no description for a facet, the value is None.
+    """
+
     es = Elasticsearch(app.app.config['ELASTICSEARCH_URL'])
     config_path = os.path.join(app.app.config['DATASET_CONFIG_DIR'], 'ui.json')
     facets_config = _parse_json_file(config_path)['facets']
 
     # Preserve order, so facets are returned in same order as the config file.
-    facets = OrderedDict()
+    es_facets = OrderedDict()
+    ui_facets = OrderedDict()
 
     for facet_config in facets_config:
         field_name = facet_config['elasticsearch_field_name']
         field_type = _get_field_type(es, field_name)
         ui_facet_name = facet_config['ui_facet_name']
+
+        ui_facets[ui_facet_name] = {'name': field_name, 'type': field_type}
```

Comment:
Sorry for not catching earlier -- can `name` be `elasticsearch_field_name`? `name` could be UI facet name. (Note even `field_name` is vague -- with upcoming variable exploration feature, it's not clear if `field_name` is Elasticsearch field name.)

Reply:
Done